File: api-cmssignature/api-mssclient/turktelekommssprovider/src/tr/gov/tubitak/uekae/esya/api/webservice/mssclient/turktelekom/signature/TurkTelekomSignatureResponse.java
package tr.gov.tubitak.uekae.esya.api.webservice.mssclient.turktelekom.signature;

import org.etsi.uri.ts102204.v1_1.turktelekom.MSSSignatureRespType;
import org.etsi.uri.ts102204.v1_1.turktelekom.SignatureType;
import org.etsi.uri.ts102204.v1_1.turktelekom.StatusType;
import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.signature.ISignatureResponse;
import tr.gov.tubitak.uekae.esya.api.infra.mobile.Status;

/**
 * Turkcell mobile signature response implementation
 * @see tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.signature.ISignatureResponse
 */
public class TurkTelekomSignatureResponse implements ISignatureResponse {
    MSSSignatureRespType mssSignatureResp;

    public TurkTelekomSignatureResponse(MSSSignatureRespType mssSignatureResp) {
        this.mssSignatureResp = mssSignatureResp;
    }

    public Status getStatus() {
        final StatusType status = mssSignatureResp.getStatus();
        if (status == null) {
            return Status.REQUEST_OK;
        }

        return new Status(status.getStatusCode().getValue().toString(), status.getStatusMessage());
    }

    public byte[] getSignature() {
        SignatureType mssSignature = mssSignatureResp.getMSSSignature();
        if(mssSignature==null)
        {
            return null;
        }
        return (byte[])mssSignature.getBase64Signature();
    }

    public byte[] getRawSignature() throws ESYAException {
        return null;
    }

    public String getTransId() {
        return mssSignatureResp.getMSSPTransID();
    }

    public String getMSISDN() {
        return mssSignatureResp.getMobileUser().getMSISDN();
    }
}

File: api-cmssignature/api-mssclient/turktelekommssprovider/csharp/src/turktelekom/signature/TurkTelekomSignatureResponse.cs
using System;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.turktelekom.stub.signature;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.signature;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper;
using SignatureType = tr.gov.tubitak.uekae.esya.api.webservice.mssclient.turktelekom.stub.signature.SignatureType;

namespace tr.gov.tubitak.uekae.esya.api.webservice.mssclient.turktelekom.signature
{
  /**
 * Turk Telekom mobile signature response implementation
 * @see ISignatureResponse
 */

    public class TurkTelekomSignatureResponse : ISignatureResponse
    {
        private readonly MSS_SignatureRespType _trcellResp;

        public TurkTelekomSignatureResponse(MSS_SignatureRespType aResponse)
        {
            _trcellResp = aResponse;
        }

        public Status getStatus()
        {
            return new Status(_trcellResp.Status.StatusCode.Value, _trcellResp.Status.StatusMessage);
        }

        public byte[] getSignature()
        {
            SignatureType mssSignature = _trcellResp.MSS_Signature;
            if(mssSignature==null)
            {
                return null;
            }
            return (byte[]) mssSignature.Base64Signature;
        }

        public byte[] getRawSignature()
        {
            return null;
        }

        public String getTransId()
        {
            return _trcellResp.MSSP_TransID;
        }

        public String getMSISDN()
        {
            return _trcellResp.MobileUser.MSISDN;
        }
    }
}